lib/logstash/outputs/azure_loganalytics.rb

```diff
@@ -84,9 +79,16 @@ def receive(event)
   public
   def flush (events, close=false)
   
-    documents = []  #this is the array of hashes to add Azure Log Analytics
+    documentsByLogType = {}  # This is a map of log_type to list of documents (themselves maps) to send to Log Analytics
     events.each do |event|
       document = {}
+      
+      log_type_for_event = event.sprintf(@log_type)
+      if log_type_for_event.match(/[^a-zA-Z0-9_]/) or log_type_for_event.length > 100
```

Comment:
`Azure::Loganalytics::Datacollectorapi::Client` do validation check on log_type before posting data to Azure. So log_type check may not be needed here

- https://github.com/yokawasa/azure-log-analytics-data-collector/blob/c9a05caac90c60048ced37282b4dd22586cb4565/lib/azure/loganalytics/datacollectorapi/client.rb#L35

Reply:
Oh, cool. I'll remove the check, thanks :)